src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/EntityAIStructureMiner.java

```diff
@@ -842,539 +601,97 @@ private boolean secureBlock(@NotNull final BlockPos curBlock, @NotNull final Blo
-        }
-        if (node.getStyle() == Node.NodeType.BEND)
-        {
-            setNodeStatusForDirection(node, direction, Node.NodeStatus.WALL);
-            int otherDirection = Math.max(direction, invertDirection(direction)) == 2 ? 4 : 2;
-            //Make Bend go to random side
-            if (Math.random() > 0.5)
-            {
-                otherDirection = invertDirection(otherDirection);
-            }
-            setNodeStatusForDirection(node, otherDirection, Node.NodeStatus.WALL);
-        }
-        //No need to do anything for CROSSROAD
-        return node;
-    }
+        workingNode = null;
 
-    private Node findNodeOnLevel(@NotNull final Level currentLevel)
-    {
-        @Nullable Node currentNode = currentLevel.getLadderNode();
-        @NotNull final LinkedList<Node> visited = new LinkedList<>();
-        while (currentNode != null)
+        if(job.getStructure() != null)
         {
-            if (visited.contains(currentNode))
-            {
-                return null;
-            }
-
-            visited.add(currentNode);
-            if (currentNode.getStatus() == Node.NodeStatus.AVAILABLE || currentNode.getStatus() == Node.NodeStatus.IN_PROGRESS)
-            {
-                return currentNode;
-            }
-
-            @NotNull final List<Integer> directions = Arrays.asList(1, 2, 3, 4);
-            Collections.shuffle(directions);
-            for (final Integer dir : directions)
-            {
-                final Node.NodeStatus status = getNodeStatusForDirection(currentNode, dir);
-                if (status == Node.NodeStatus.AVAILABLE || status == Node.NodeStatus.IN_PROGRESS)
-                {
-                    return currentNode;
-                }
-                if (status == Node.NodeStatus.COMPLETED)
-                {
-                    final Optional<Node> first = tryFindNodeInDirectionOfNode(currentLevel, currentNode, dir);
-                    if (first.isPresent())
-                    {
-                        if (visited.contains(first.get()))
-                        {
-                            //Stop endless loops
-                            continue;
-                        }
-                        if (getNodeStatusForDirection(first.get(), invertDirection(dir)) == Node.NodeStatus.WALL)
-                        {
-                            //We got to a wall, not useful
-                            continue;
-                        }
-                        currentNode = first.get();
-                        //Out of direction for loop
-                        break;
-                    }
-
-                    @NotNull final Node newnode = createNewNodeInDirectionFromNode(currentNode, dir);
-                    currentLevel.addNode(newnode);
-                    return newnode;
-                }
-            }
+            return CLEAR_STEP;
         }
 
-        return null;
+        return AIState.MINER_MINING_NODE;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Unnecessary use of fully qualified name 'AIState.MINER_MINING_NODE' due to existing static import 'com.minecolonies.coremod.entity.ai.util.AIState.*' [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3AUnnecessaryFullyQualifiedName)